packages/block-editor/src/components/link-control/index.js

```diff
@@ -77,9 +81,25 @@ function LinkControl( {
 	};
 
 	// Utils
-	const startEditMode = () => {
-		if ( isFunction( onLinkChange ) ) {
-			onLinkChange();
+
+	/**
+	 * Handler function which switches the mode of the component,
+	 * between `edit` and `show` mode.
+	 * Also, it calls `onChangeMode` callback function.
+	 *
+	 * @param {string} mode Component mode: `show` or `edit`.
+	 */
+	const setMode = ( mode = 'show' ) => () => {
+		setIsEditingLink( MODE_EDIT === mode );
+
+		// Populate input searcher whether
+		// the current link has a title.
+		if ( currentLink && currentLink.title ) {
+			setInputValue( currentLink.title );
+		}
```

Comment:
If we need this to be an "initializer" for the input value, couldn't we instead provide this as the initial value for the input state?

In other words, instead of:

https://github.com/WordPress/gutenberg/blob/84a46599bdc815b0c8a2781978486426b14b54c5/packages/block-editor/src/components/link-control/index.js#L59

Do:

```js
const [ inputValue, setInputValue ] = useState( value.title );
```

Reply:
See: #19737

Reply:
It makes totally sense to me, Andrew. I don't exactly the reason why we added this initializer, especially because this component has been significantly refactored lately.
Going to help to review at your PR.